File: RotateMatrix/rotate_matrix.js
const rotate = (matrix) => {
  console.log("Input Matrix: ", matrix);
  matrix.reverse();

  for (let i = 0; i < matrix.length; i++) {
    for (let j = 0; j < i; j++) {
      console.log(i, j);
      let temp = matrix[i][j];
      matrix[i][j] = matrix[j][i];
      matrix[j][i] = temp;
    }
  }

  console.log("Rotated Matrix: ", matrix);
};

const run = (() => {
  const matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [21, 48, 59],
  ];
  rotate(matrix);
})();

File: RotateMatrix/rotate_matrix.rs
fn main() {
  let mut matrix: Vec<Vec<i32>> = vec![vec![1, 2, 3], vec![4, 5, 6], vec![21, 48, 59]];
  println!("Input Matrix: {:?}", matrix);

  rotate_matrix(&mut matrix);
  println!("Output Matrix: {:?}", matrix);
}


fn rotate_matrix(matrix: &mut Vec<Vec<i32>>) {
  matrix.reverse();

  for i in 0..matrix.len() {
    for j in 0..matrix.len() {
      if i < j {
        let temp = matrix[j][i];
        matrix[j][i] = matrix[i][j];
        matrix[i][j] = temp;
      }
    }
  }  
}